### PR TITLE
PACT-871 : Modify request and response for creators to match new API schema

### PR DESCRIPTION
**Pull request by userjd**

### CODE REVIEW

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  val displayedName = {
+    val dateDisplay = (agentType, dateFrom, dateTo) match {
+      case (AgentType.CorporateBody, Some(dateFrom), Some(dateTo)) => s" ($dateFrom - $dateTo)"
+      case (AgentType.CorporateBody, Some(dateFrom), None)         => s" ($dateFrom - )"
+      case (AgentType.CorporateBody, None, Some(dateTo))           => s" ( - $dateTo)"
+      case (AgentType.Person, Some(dateFrom), Some(dateTo))        => s" (b.$dateFrom - d.$dateTo)"
+      case (AgentType.Person, Some(dateFrom), None)                => s" (b.$dateFrom - )"
+      case (AgentType.Person, None, Some(dateTo))                  => s" ( - d.$dateTo)"
+      case (_, None, None)                                         => ""
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala:42]: `case _ => ""` might be better here.

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+case class AgentSummary(
+  agentType: AgentType,
+  identifier: String,
+  label: String,
+  dateFrom: Option[Int] = None,
+  dateTo: Option[Int] = None
+) {
+
+  val displayedName = {
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala:34]: Missing type annotation.

**Review by rwalpole — CHANGES_REQUESTED** [review]
We are missing a test for `getAgentSummaries` in `MessagingServiceSpec`. In fact there should be at least two - one for the happy path and one where the reply message cannot be parsed.

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala (hunk 1 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala]: Done

File: app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala (hunk 2 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/models/AgentSummary.scala]: Done

**Review by rwalpole — APPROVED** [review]

**Pull request merged**